loan-payoff/REPWRITERSPECS/BANNO.LOAN.PAYOFF.V1.POW

```diff
@@ -833,6 +867,84 @@ PROCEDURE DISPLAYRESULTS
  PRINT "},"
  NEWLINE
  CALL GETCOLLATERAL
+
+ IF PAYOFFFEECOUNT>0 THEN
+  DO
+   PRINT "   "+Q+"payoffFees"+Q+":["
+   NEWLINE
+   FOR TMPLOOP=1 TO PAYOFFFEECOUNT
+    DO
+     TMPCHR=FORMAT("#######9.99",PAYOFFFEEAMT(TMPLOOP))
+     CALL REMOVELEADINGSPACES
```

Comment:
(Realize that this is not relevant to your coding changes) - I noticed that there are two procedures for removing leading spaces (REMOVELEADINGSPACES & TMPCHRNLS) Can we choose one to use and get rid of the other?  Also, for the related procedure TMPCHRNTS... can that just be reduced to

```
PROCEDURE TMPCHRNTS
 TMPCHR=SEGMENT(TMPCHR,1,LENGTH(TMPCHR))
END
```

Reply:
I'm not sure about the TMPCHRNTS question.  Does the LENGTH function not consider trailing spaces when determining the length?  Your proposed change does simplify things quite a bit if the above is true.

On REMOVELEADINGSPACES and TMPCHRNLS, it would be nice to eliminate one.  Those procedures, logically, do approach the removing of leading spaces a little different.  Do we feel comfortable that they produce the same results?  I don't recall the specifics but seems like there was something weird/tricky about how leading/trailing spaces and the LENGTH function.  So I guess I am on the fence with regards to changing this code.  What are your thoughts?

Reply:
In walking through the two variations of the code to remove leading spaces, it seems that they're both ending up with the same results - albeit it, like you noted, in slightly different ways.  The LENGTH statement does not take into consideration any trailing spaces (according to eDocs - `Counts leading, not trailing, spaces in the length`). Thus "`Tom   '" (which has three spaces at the end) should be returning a length of 3. Also, because both versions of the procedures to remove leading spaces reference the LENGTH function, they also, inadvertently, removing trailing spaces as well. 

In looking at the code, however for the two TMPCHR... procedures, I'm thinking that they both might fail IF the value they're testing is all spaces.  Given a TMPCHR value of `"   "` (three spaces) each of those procedures first start by testing for a LENGTH(TMPCHR) being greater than 0.  Since the LENGTH function does not count trailing spaces, in this case it "should" return a length of 0. Since it fails the LENGTH(TMPCHR)>0 test, all the code in the DO/END clause should be getting skipped over and thus, TMPCHR would get left with a value of 3 spaces... and thus, would also not be blank... technically.

Reply:
I think it'd be best to clean it up a little and take one out given how eDocs says the LENGTH function works. But it also goes back to Robby's previous note that there's a lot that can be re-worked in the original code. As far as it failing on spaces, I think it'd be best to just double check where the procedures are called and ensure they'll have data besides a space. It's probably not an issue since they seem to be primarily used to remove spaces after a format, and all the formats have 9's, so at least a zero will be passed.